dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -224,51 +214,53 @@ public void updateTranslations( IdentifiableObject persistedObject, Set<Translat
 
     @Override
     @Transactional
-    public void delete( IdentifiableObject object )
+    public void delete( @Nonnull IdentifiableObject object )
     {
         delete( object, currentUserService.getCurrentUser() );
     }
 
     @Override
     @Transactional
-    public void delete( IdentifiableObject object, User user )
+    public void delete( @Nonnull IdentifiableObject object, @CheckForNull User user )
     {
-        IdentifiableObjectStore<? super IdentifiableObject> store = getIdentifiableObjectStore(
-            HibernateProxyUtils.getRealClass( object ) );
+        IdentifiableObjectStore<? super IdentifiableObject> store = getIdentifiableObjectStore( object );
 
         if ( store != null )
         {
             store.delete( object, user );
         }
     }
 
+    @Nonnull
     @Override
     @Transactional( readOnly = true )
-    public Optional<? extends IdentifiableObject> find( String uid )
+    public Optional<? extends IdentifiableObject> find( @Nonnull String uid )
     {
         return identifiableObjectStores.stream()
             .map( store -> store.getByUid( uid ) )
             .filter( Objects::nonNull )
             .findFirst();
     }
 
+    @Nonnull
     @Override
     @Transactional( readOnly = true )
-    public <T extends IdentifiableObject> T get( Class<T> type, long id )
+    public <T extends IdentifiableObject> T get( @Nonnull Class<T> type, long id )
     {
         IdentifiableObjectStore<T> store = getIdentifiableObjectStore( type );
 
         if ( store == null )
         {
-            return null;
+            throw new ObjectNotFoundException( id, "No store registered for objects of type: " + type );
```

Comment:
OBS! Either this method should return an object or throw an exception. This makes it consistent.